### ISSUE TITLE
Liquid: Indent Comments Not Working

### ISSUE THREAD

**Issue by macdonaldr93** [prettydiff/prettydiff#283]
Looks like comments aren't being detected in liquid. I'm using `ERB Template` to beautify the liquid code. 

## Source

```
{% comment %}
  Sed aliquam ultrices mauris. Donec quam felis, ultricies nec, 
  pellentesque eu, pretium quis, sem. Nullam accumsan lorem in 
  dui. Quisque id odio. Praesent porttitor, nulla vitae posuere 
  iaculis, arcu nisl dignissim dolor, a pretium mi sem ut ipsum.

  {% include 'lorem-ipsum' %}
{% endcomment %}
```

### No comment indentation

```
{% comment %}
Sed aliquam ultrices mauris. Donec quam felis, ultricies nec, pellentesque eu,
pretium quis, sem. Nullam accumsan lorem in dui. Quisque id odio. Praesent
porttitor, nulla vitae posuere iaculis, arcu nisl dignissim dolor, a pretium mi
sem ut ipsum.

{% include 'lorem-ipsum' %}
{% endcomment %}
```

### Indent comments

```
{% comment %}
Sed aliquam ultrices mauris. Donec quam felis, ultricies nec, pellentesque eu,
pretium quis, sem. Nullam accumsan lorem in dui. Quisque id odio. Praesent
porttitor, nulla vitae posuere iaculis, arcu nisl dignissim dolor, a pretium mi
sem ut ipsum.

{% include 'lorem-ipsum' %}
{% endcomment %}
```

should be:

```
{% comment %}
  Sed aliquam ultrices mauris. Donec quam felis, ultricies nec, 
  pellentesque eu, pretium quis, sem. Nullam accumsan lorem in 
  dui. Quisque id odio. Praesent porttitor, nulla vitae posuere 
  iaculis, arcu nisl dignissim dolor, a pretium mi sem ut ipsum.

  {% include 'lorem-ipsum' %}
{% endcomment %}
```

**Comment by prettydiff** [comment]
Resolve on my local.  I will publish later today.

**Comment by macdonaldr93** [comment]
Amazing! Thank you. Let me know when you want me to test.

**Comment by prettydiff** [comment]
Try it out at http://prettydiff.com/ignore/testlocation/index.xhtml?m=beautify

**Comment by macdonaldr93** [comment]
Just moving the conversation about comments over here :)

> The comments are preserved verbatim exactly like an HTML. This includes absolutely all whitespace and the end comment tag. 

## Output

```
{% if cat != tag and cat_item == cat %}
    {% comment %}
    Curabitur vestibulum aliquam leo. Nam commodo suscipit quam
  {% endcomment %}
    {% if current_tags contains tag %}
        <li
            class="lorem-filter active-filter"
            data-group="{{ cat_item }}"
            data-handle="{{ tag | handle }}">{{ tag | remove_first: cat_item | remove_first: '_' | link_to_remove_tag: tag }}</li>
    {% else %}
        <li
            class="lorem-filter"
            data-group="{{ cat_item }}"
            data-handle="{{ tag | handle }}">{{ tag | remove_first: cat_item | remove_first: '_' | link_to_add_tag: tag }}</li>
    {% endif %}
{% endif %}
```

The only difference I really expected was for the `{% endcomment %}` tag to be aligned with the `{% comment %}` tag.

**Comment by prettydiff** [comment]
I can do that.  At this time the comment start tag, comment body, and comment end tag are captured as a single unified tag with 100% preservation.  Splitting these three into separate parsed tokens will look much better and allow a finer level of manipulation and analysis, but it will also mean manipulation to the internal white space.

I will get this worked up in the next few hours.

**Comment by macdonaldr93** [comment]
Amazing. I think having the flexibility will be worth it. You're killing it! Thanks so much for this.

**Comment by prettydiff** [comment]
Comments adjusted: http://prettydiff.com/ignore/testlocation/index.xhtml?m=beautify

**Comment by macdonaldr93** [comment]
sweetness, works great!

**Comment by prettydiff** [comment]
Released with version v1.16.28.
